.circleci/config.yml

```diff
@@ -0,0 +1,175 @@
+version: 2.1
+
+orbs:
+  python: circleci/python@2.0.3
+
+# Further ideas for jobs to run:
+# - license check
+# - make sure test coverage increases
+# And, once we're sure that the pipeline is working:
+# - integration/performance tests
+# Once we have docs:
+# - documentation coverage
+# - docs build (on master only)
+
+
+workflows:
+  mite:
+    jobs:
+      - check-acurl-version
+      - build-and-test:
+          requires:
+            - check-acurl-version
+      - tag:
+          requires:
+            - build-and-test
+            - check-acurl-version
+          filters:
+            branches:
+              only: master
+      - linux-wheels:
+          requires:
+            - tag
+          filters:
+            branches:
+              only: master
+      - osx-wheels:
+          requires:
+            - tag
+          filters:
+            branches:
+              only: master
+
+
+jobs:
+  check-acurl-version:
+    docker:
+      - image: cimg/python:3.10
+    steps:
+      - checkout
+      - run:
+          command: |
+            ACURL_CHANGED=false
+
+            if [[ -n $(git show --name-only ${CIRCLE_SHA1} ./acurl) ]]; then
+              echo "Acurl has changed"
+              ACURL_CHANGED=true
+
+              LATEST_ACURL_VERSION=$(curl -s https://pypi.org/pypi/acurl/json | jq -r '.info .version')
+              if [[ -n $(grep "version = $LATEST_ACURL_VERSION" acurl/setup.cfg) ]]; then
+                echo "Acurl has changed, but the version of acurl hasn't changed"
+                exit 1
+              fi
+            fi
+
+            mkdir -p /tmp/workspace
+            echo "export ACURL_CHANGED=\"$ACURL_CHANGED\"" >> /tmp/workspace/env_vars
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - env_vars
+
+  build-and-test:
+    docker:
+      - image: cimg/python:3.10
+    steps:
+      - checkout
+      - python/install-packages:
+          pkg-manager: pip
+      - python/install-packages:
+          pip-dependency-file: test-requirements.txt
+          pkg-manager: pip
+      - python/install-packages:
+          pip-dependency-file: dev-requirements.txt
+          pkg-manager: pip
+      - run:
+          name: pre-commit checks
+          command: pre-commit run --origin HEAD --source origin/master || git diff
+      - run:
+          name: tox
+          command: tox -e py310
```

Comment:
do we care about also running the tests under other python versions?  i think we've caught the occasional compatibility error that way.  that's supposed to be what tox supports.  admittedly the old test pipeline could not do this, but that's due to the way it was shackled to a jenkins image that made it very difficult to install multiple pythons.  now we could either use a multi-python image to run the tests in (if such a thing is available), or else have multiple test jobs for 3.9 and 3.10.

Reply:
I can look into it. I don't think there's a multi-python image, but I'll check if there's a good/easy method for setting one up (i.e. using a python 3.10 image and install 3.8 & 3.9 along side). or go the easy route and like you suggest, multiple jobs.

Reply:
I tried using pyenv, but couldn't get it working reliably. I've gone for multiple jobs instead, but I'll keep it on my list of improvements to make, as ideally it should be one job.